enhancements/console/customize-add-page.md

```diff
@@ -0,0 +1,174 @@
+
+A final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    addPage:
+      hideGroups:
+      - developer-catalog
+      - git-repository
+      - container-images
+      - source-files
+      - serverless
+      - pipelines
+      hideActions:
+      - databases
+      - helmcharts
+      - operatorbacked
+      - virtualmachinetemplates
+      - managedservices
+      - eventsources
+      - fromgit
+      - fromdevfile
+      - fromdockerfile
+      - containerimage
+      - importyaml
+      - uploadajar
+      - eventsource
+      - channel
+      - pipelines
+  ...
+```
+
+### Provide a code snippet in YAML editor
+
+The admin needs to know the current available groups and actions to customize them. We don't want to provide the defaults in the CR so that we can change these defaults later without migrating the customer data.
+
+To provide the default values close to the editor, the console will provide two code snippets in the sidebar of the YAML editor when editing the `Console` operator config. We can provide one snippet for the groups and one for the actions. The snippets should include the ids of all extensions.
+
+### User Stories
+
+#### Story 1
+
+As an cluster admin, I want to customize the add page. The admin needs to know the current list of available groups and actions. The admin can configure a list of groups and actions which should not be shown on the add page.
+
+#### Story 2
+
+As a user, I don't want to see any groups or actions on the add page which are hidden by the cluster admin.
+
+### Risks and Mitigations
+
+#### How we handle upgrade
+
+**Mitigation**: The cluster admin can provide a list of groups and actions which should not shown to the users. All features which are not defined in that list will be shown after an upgrade.
+
+**Drawback** If the cluster admin installs a new OpenShift upgrade the customized add page will contain new or unexpected actions.
+
+## Open Questions
+
+- [x] Should we save a list of enabled groups and actions instead of a list of ids which should not be shown?
+
+For example:
+
+```yaml
+spec:
+  customization:
+    addPage:
+      groups:
+      - developer-catalog
+      - git-repository
+      - ...
+      actions:
+      - databases
+      - helmcharts
+      - ...
+  ...
+```
+
+Answer:
+
+> Customers requested to hide some unnecessary actions and add their own actions to the add page. Product management expected that it's easier for them to define a negative list, so that new actions will be shown automatically if an operator is installed.
+
+- [ ] If we save a list of shown and hidden actions, its maybe easier later to show an admin the information which new actions are available after an upgrade or when a new operator is installed.
```

Comment:
Haven't seen this requirement from UX and PM.

Reply:
Ack! Checked this question and added an answer

> No. This is not required at the moment and we will consider such requirements again when we will provide an UI for the admin in the future.